### PR TITLE
A ref type's subtype is a context in which only a type is expected.

### LINKED ISSUE #14174: ref returns and locals have problems with binding  "ref dynamic"
I suspect it could be result of the recent refactoring of ref returns/locals syntax.
Note that   ref dynamic parameters are ok.

```cs

   class Program
    {
        static object[] arr = new object[] { "a" };

        static void Main(string[] args)
        {
            RefParam(ref arr[0]);
            System.Console.WriteLine(x.ToString());


            // error
            ref dynamic x = ref arr[0];
            x = "b";

            System.Console.WriteLine(x.ToString());

            RefReturn() = "e";
            System.Console.WriteLine(x.ToString());

        }

        // works fine
        static void RefParam(ref dynamic p)
        {
            p = "c";
        }

        // error
        static ref dynamic RefReturn()
        {
            return ref arr[0];
        }
    }
```

error CS0246: The type or namespace name 'dynamic' could not be found (are you missing a using directive or an assembly reference?)

NOTE: I tried to make a repro similar to https://github.com/dotnet/roslyn/issues/14166 , but with dynamic, not with tuples.
I think we might have same issue with dynamic and my fix for #14166 would fix it, but that is theoretical, since I can't make "ref dynamic" to work at all in these contexts.

### PR DESCRIPTION
**Pull request by gafter**
Fixes #14174 

@VSadov @dotnet/roslyn-compiler Please review this tiny bug fix.

### CODE REVIEW

File: src/Compilers/CSharp/Test/Semantic/Semantics/RefLocalsAndReturnsTests.cs [hunk 1]
```diff
@@ -960,5 +960,43 @@ void F(object[] a, object[,] a2, int i)
...
+
+    static ref dynamic RefReturn()
+    {
+        return ref arr[0];
+    }
+}
+";
+            CompileAndVerify(text,
+                expectedOutput: "frog",
+                additionalRefs: new[] { SystemCoreRef, CSharpRef }).VerifyDiagnostics();
```
Comment (VSadov) [inline src/Compilers/CSharp/Test/Semantic/Semantics/RefLocalsAndReturnsTests.cs]: :-)

**Comment by VSadov** [comment]
LGTM

**Comment by mattwar** [comment]
@dotnet-bot test windows_debug_unit32_prtest please

**Comment by AlekseyTs** [comment]
LGTM

**Pull request merged**